src/aio/native_linux_aio_provider.cpp

```diff
@@ -68,14 +70,33 @@ error_code native_linux_aio_provider::flush(dsn_handle_t fh)
 error_code native_linux_aio_provider::write(const aio_context &aio_ctx,
                                             /*out*/ uint32_t *processed_bytes)
 {
-    ssize_t ret = pwrite(static_cast<int>((ssize_t)aio_ctx.file),
-                         aio_ctx.buffer,
-                         aio_ctx.buffer_size,
-                         aio_ctx.file_offset);
-    if (ret < 0) {
-        return ERR_FILE_OPERATION_FAILED;
+    int retries = 2;
+    uint32_t buffer_offset = 0;
+    while (--retries >= 0) {
+        uint32_t ret = pwrite(static_cast<int>((ssize_t)aio_ctx.file),
+                              (char *)aio_ctx.buffer + buffer_offset,
+                              aio_ctx.buffer_size - buffer_offset,
+                              aio_ctx.file_offset + buffer_offset);
+        if (ret < 0) {
+            return ERR_FILE_OPERATION_FAILED;
```

Comment:
Why don't we try again here?

Reply:
1. This pr is not fix for `write failed` but only `write incompletely`
2. This bug is not easy to repeat, and it only decrease the probability but not resolve it completely